command/job_start_test.go

```diff
@@ -0,0 +1,200 @@
+package command
+
+import (
+	"encoding/json"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/posener/complete"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/nomad/ci"
+	"github.com/hashicorp/nomad/command/agent"
+	"github.com/hashicorp/nomad/helper/pointer"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/mitchellh/cli"
+	"github.com/shoenig/test/must"
+)
+
+var _ cli.Command = (*JobStartCommand)(nil)
+
+func TestJobStartCommand_Fails(t *testing.T) {
+	ci.Parallel(t)
+
+	srv, _, addr := testServer(t, true, func(c *agent.Config) {
+		c.DevMode = true
+	})
+	defer srv.Shutdown()
+
+	ui := cli.NewMockUi()
+	cmd := &JobStartCommand{Meta: Meta{Ui: ui}}
+
+	// Fails on misuse
+	code := cmd.Run([]string{"-bad", "-flag"})
+	must.One(t, code)
+
+	out := ui.ErrorWriter.String()
+	must.StrContains(t, out, "flag provided but not defined: -bad")
+
+	ui.ErrorWriter.Reset()
+
+	// Fails on nonexistent job ID
+	code = cmd.Run([]string{"-address=" + addr, "non-existent"})
+	must.One(t, code)
+
+	out = ui.ErrorWriter.String()
+	must.StrContains(t, out, "No job(s) with prefix or ID")
+
+	ui.ErrorWriter.Reset()
+
+	// Fails on connection failure
+	code = cmd.Run([]string{"-address=nope", "n"})
+	must.One(t, code)
+
+	out = ui.ErrorWriter.String()
+	must.StrContains(t, out, "Error querying job prefix")
+
+	// Fails on attempting to start a job that's not been stopped
+	jobID := uuid.Generate()
+	jobFilePath := filepath.Join(os.TempDir(), jobID+".nomad")
+
+	t.Cleanup(func() {
+		_ = os.Remove(jobFilePath)
+	})
+	job := testJob(jobID)
+	job.TaskGroups[0].Tasks[0].Resources.MemoryMB = pointer.Of(16)
+	job.TaskGroups[0].Tasks[0].Resources.DiskMB = pointer.Of(32)
+	job.TaskGroups[0].Tasks[0].Resources.CPU = pointer.Of(10)
+	job.TaskGroups[0].Tasks[0].Config = map[string]interface{}{
+		"run_for": "30s",
+	}
+
+	jobJSON, err := json.MarshalIndent(job, "", " ")
+	must.NoError(t, err)
+
+	jobFile := jobFilePath
+	err = os.WriteFile(jobFile, []byte(jobJSON), 0o644)
+	must.NoError(t, err)
+
+	runCmd := &JobRunCommand{Meta: Meta{Ui: ui}}
+	code = runCmd.Run([]string{"-address", addr, "-json", jobFile})
+	must.Zero(t, code,
+		must.Sprintf("job stop stdout: %s", ui.OutputWriter.String()),
+		must.Sprintf("job stop stderr: %s", ui.ErrorWriter.String()),
+	)
+
+	code = cmd.Run([]string{"-address=" + addr, jobID})
+	must.One(t, code)
+	out = ui.ErrorWriter.String()
+	must.StrContains(t, out, "has not been stopped and has following status:")
```

Comment:
If we decide to not return an error when a user attempts to start a running job, this final test for misuse might not be necessary here.